.github/workflows/e2e-tests.yml

```diff
@@ -114,6 +114,26 @@ jobs:
               mv src/Baz.php.orig src/Baz.php
               echo -n > phpstan-baseline.neon
               ../../bin/phpstan -vvv
+          - script: |
+              cd e2e/result-cache-6
+              echo -n > phpstan-baseline.neon
+              ../../bin/phpstan -vvv
+              patch -b src/Baz.php < patch-1.patch
+              cat baseline-1.neon > phpstan-baseline.neon
+              ../../bin/phpstan -vvv
+              mv src/Baz.php.orig src/Baz.php
+              echo -n > phpstan-baseline.neon
+              ../../bin/phpstan -vvv
+          - script: |
+              cd e2e/result-cache-8
+              echo -n > phpstan-baseline.neon
+              ../../bin/phpstan -vvv
+              patch -b src/Baz.php < patch-1.patch
+              cat baseline-1.neon > phpstan-baseline.neon
+              ../../bin/phpstan -vvv
+              mv src/Baz.php.orig src/Baz.php
+              echo -n > phpstan-baseline.neon
+              ../../bin/phpstan -vvv
```

Comment:
I tried `result-cache-8` after removing changes from `src/` and it does not fail for me locally. Please first submit a draft PR with just the E2E tests so it's obvious they fail.

Reply:
after having slept about it, I think the situation is as follows.

we only need to support `require-extends` for result-cache invalidation because thats the only new part in the [reflection hierarchy](https://github.com/phpstan/phpstan-src/pull/2856/files#diff-9007a149012fc95c1311cb1346c8e75f59786c0aafb5390f2e56d28f0bc2ff8a). 

`require-implements` works with reflection information which was available before, we just added some rules arround it.

[see failling test](https://github.com/phpstan/phpstan-src/actions/runs/7499547666/job/20416549891?pr=2866)
[and the fix](https://github.com/phpstan/phpstan-src/pull/2866/commits/2282c7e8170744e677d87573917d6411b91214fb)